### PR TITLE
Expose chunked_threshold on Response. Fix #149

### LINKED ISSUE #149: Allow non-chunked transfer for larger files
Hi!

We're happy users of Rouille, however when sending larger files, we hit this issue:

https://github.com/tiny-http/tiny-http/blob/master/src/response.rs#L167-L168

```
    // if we don't have a Content-Length, or if the Content-Length is too big, using chunks writer
    let chunks_threshold = 32768;
    if entity_length.as_ref().map_or(true, |val| *val >= chunks_threshold) {
        return TransferEncoding::Chunked;
    }
```

This makes for a bad user experience, since they don't get any progress meter on downloads in the browser.

Would you consider a PR that lets us configure this behaviour? I.e. maybe raising the limit on a per-response basis?

### PR DESCRIPTION
**Pull request by lolgesten**

### CODE REVIEW

**Comment by lolgesten** [comment]
@tomaka @frewsxcv I apologise for pinging you, I know you're probably both very busy.

Does anyone have any opinion on this issue? Does it go against the design goals? Can it be merged?

File: src/response.rs [hunk 1]
```diff
@@ -278,7 +297,8 @@ impl<R> Response<R> where R: Read {
                                -> IoResult<()>
     {
         let mut transfer_encoding = Some(choose_transfer_encoding(request_headers,
-                                    &http_version, &self.data_length, false /* TODO */));
+                                    &http_version, &self.data_length, false,
+                                    self.chunked_threshold() /* TODO */));
```
Comment (tomaka) [inline src/response.rs]: The `TODO` is applied to `false`, so please move it in the line above.

File: src/response.rs [hunk 2]
```diff
@@ -207,6 +209,22 @@ impl<R> Response<R> where R: Read {
         response
     }
 
+    /// Set a threshold for `Content-Length` where we chose chunked
+    /// transfer. Notice that chunked transfer might happen regardless of
+    /// this threshold, for instance when the request headers indicate
+    /// it is wanted or when there is no `Content-Length`.
+    pub fn set_chunked_threshold(&mut self, length: usize) {
```
Comment (tomaka) [inline src/response.rs]: Should be `with_chunked_threshold`.

**Comment by lolgesten** [comment]
Hi @tomaka did my update address your concerns?

**Comment by lolgesten** [comment]
@tomaka Sorry to chase. Is there anything needed to land this?

**Pull request merged**